torch/_subclasses/meta_utils.py

```diff
@@ -523,6 +530,24 @@ def __call__(
                 # instrumentation will see the meta conversions and the
                 # tests all break so we just exclude this.  In any case
                 # the to conversion isn't really right anyhow.
+
```

Comment:
FYI @bdhirsh - we're doing something that morally should just be adding a tensor_unflatten / tensor_flatten to FunctionalTensorWrapper, but since those APIs don't exist yet we're just hardcoding this into the fakeification logic. Curious if you think this is the right place to toss it or if you were putting the tensor_unflatten/tensor_flatten handling somewhere else

Reply:
Is the goal of this PR specifically to get `torch.compile` working on code that runs inside of a `functorch.functionalize()` call? Or is it to more generically support functionalization wherever its used in the stack (functionalization kernel for torch.cond, and functionalization that runs in AOTAutograd).

I'm asking because AOTAutograd is (hopefully) soon going to start using a python version of `FunctionalTensorWrapper` ([PR](https://github.com/pytorch/pytorch/pull/106405/files#diff-a578c8c54653bc4c7b0775ad273154ac7e9ece893f6bf2fda1874c088b16072aR1)). Although I haven't given it a flatten/unflatten API (mostly because in AOTAutograd, I don't actually **want** to flatten it).

I wonder if you think it'd be worth just handling this logic for the python `FunctionalTensor` and not bothering for the C++ variant. That'd probably require updating torch.cond's functionalization rule to use the python flavor. And.. I'm also not sure if we can easily migrate `functorch.functionalize`'s version to use the python flavor.

But either way, I think this probably a reasonable place to put this code.